### PR TITLE
fix(pre-commit): Gitlab Custom RegistryUrl not honored

### PR DESCRIPTION
**Pull request by gone-for-coding**
## Changes

The pre-commit manager does not return custom registryUrls to the gitlab-tags datasource which then always defaults to gitlab.com.

This PR always returns a registryUrl back to the datasource which fixes the behaviour. The tests have been edited accordingly to pass.

## Context

Fixes #25157

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
I'd like to understand why other parts of this function return `registryUrls: [hostname]` and it seems to work. I had though that registryUrls must start with `https://` so wonder if they're being massaged somewhere else

**Comment by gone-for-coding** [comment]
Yes, the https:// prefix ist stripped here:https://github.com/renovatebot/renovate/blob/9a55d753f976ad82dd3ca49267da07e27e2a0fc5/lib/modules/manager/pre-commit/extract.ts#L83 So I readd it as the gitlab-tags datasource seems to require the https:// prefix to work properly.

Regarding the other parts of the function which return registryUrls, they have the https:// prefix as it is added here: https://github.com/renovatebot/renovate/blob/9a55d753f976ad82dd3ca49267da07e27e2a0fc5/lib/modules/manager/pre-commit/extract.ts#L39

As the function already returns before line 39 I needed to add it.

Currently I'm wondering if anything below line 39 is actually every executed.

File: lib/modules/manager/pre-commit/extract.ts [hunk 1]
```diff
@@ -34,7 +34,10 @@ function determineDatasource(
   }
   if (hostname === 'gitlab.com' || detectPlatform(repository) === 'gitlab') {
     logger.debug({ repository, hostname }, 'Found gitlab dependency');
-    return { datasource: GitlabTagsDatasource.id };
+    return {
+      datasource: GitlabTagsDatasource.id,
+      registryUrls: ['https://' + hostname],
```
Comment (viceice) [inline lib/modules/manager/pre-commit/extract.ts:39]: only set registry url If it's not gitlab.com

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/pre-commit/extract.ts (hunk 1 above)
Comment (gone-for-coding) [reply lib/modules/manager/pre-commit/extract.ts:46]: Hi  @viceice 
Thanks for the review. I implemented the changes and added some tests. Please have a look.

**Review by viceice — APPROVED** [review]
LGTM but you need to fix the git author and then force push changes

**Review by rarkins — APPROVED** [review]

**Pull request merged**